internal/services/signalr/signalr_service_resource.go

```diff
@@ -73,11 +73,26 @@ func resourceArmSignalRServiceCreate(d *pluginsdk.ResourceData, meta interface{}
 	}
 
 	sku := d.Get("sku").([]interface{})
-	featureFlags := d.Get("features").(*pluginsdk.Set).List()
-	connectivityLogsEnabled := d.Get("connectivity_logs_enabled").(bool)
-	messagingLogsEnabled := d.Get("messaging_logs_enabled").(bool)
-	liveTraceEnabled := d.Get("live_trace_enabled").(bool)
-	serviceMode := d.Get("service_mode").(string)
+	var featureFlags []interface{}
+	if d.Get("features") != nil {
+		featureFlags = d.Get("features").(*pluginsdk.Set).List()
+	}
```

Comment:
we could make use of the ThreePointOhBeta flag here and move this to the if/else below, something like:

```
	expandedFeatures := make([]signalr.SignalRFeature, 0)
	if !features.ThreePointOh() {
		if featureFlags := d.Get("features").(*pluginsdk.Set).List(); len(featureFlags) > 0 {
			expandedFeatures = *expandSignalRFeatures(featureFlags)
		}
	} else {
		expandedFeatures = append(expandedFeatures, signalRFeature(signalr.FeatureFlagsEnableConnectivityLogs, strconv.FormatBool(connectivityLogsEnabled)))
		expandedFeatures = append(expandedFeatures, signalRFeature(signalr.FeatureFlagsEnableMessagingLogs, strconv.FormatBool(messagingLogsEnabled)))
		expandedFeatures = append(expandedFeatures, signalRFeature("EnableLiveTrace", strconv.FormatBool(liveTraceEnabled)))

		if serviceMode == "" {
			serviceMode = "Default"
		}
		expandedFeatures = append(expandedFeatures, signalRFeature(signalr.FeatureFlagsServiceMode, serviceMode))
	}
```